chain/epoch-manager/src/types.rs

```diff
@@ -142,12 +142,25 @@ impl EpochInfoAggregator {
             // NOTE:(#11900): If the chunk endorsements received from the chunk validators are not recorded in the block header,
             // we use the chunk production stats as the endorsements stats, ie. if the chunk is produced then we assume that
             // the endorsements from all the chunk validators assigned to that chunk are received (hence the `else` branch below).
-            let chunk_endorsements =
-                if let Some(chunk_endorsements) = block_info.chunk_endorsements() {
+            let chunk_endorsements = if let Some(chunk_endorsements) =
+                block_info.chunk_endorsements()
+            {
+                // For old chunks, we optimize the block and its header by not including the chunk endorsements and
+                // corresponding bitmaps. Thus, we expect that the bitmap is non-empty for new chunks only.
+                if *mask {
+                    debug_assert!(chunk_endorsements.len(shard_id).unwrap() >= chunk_validators.len(),
```

Comment:
Looks like we can assert `chunk_endorsements.len(shard_id).unwrap() == chunk_validators.len().div_ceil(8) * 8` as above.

Reply:
done